ddprof-ffi/src/exporter.rs

```diff
@@ -400,4 +414,16 @@ mod test {
             }
         }
     }
+
+    #[test]
+    fn byte_slice_to_buffer_conversion() {
+        let raw: &[u8] = b"dummy data";
+        let slice = Slice::new(raw.as_ptr(), raw.len());
+
+        let converted: Box<Buffer> = unsafe { ddprof_ffi_Buffer_from_byte_slice(slice) };
+        let converted_contents: &[u8] = unsafe { (*converted).as_slice() };
+
+        assert_eq!(b"dummy data", converted_contents);
```

Comment:
Perhaps the test should be done once the raw data is dropped to ensure we are reading correctly off of the copy ?

Reply:
Good point. I don't think it makes sense to drop the `raw` data since it's not being heap allocated, but in 3bf0fc6fed440a9a524fe25a75ad134b30b56c3f I enhanced the test so that the raw data gets mutated and we validate that the copy is unchanged.